### PR TITLE
JSON Schema - include latest spec updates around CredSchema/digestSRI

### PR DESCRIPTION
**Pull request by decentralgabe**
- Add enforcement of credential schema for json schema credentials and associated tests

### CODE REVIEW

File: credential/schema/vcjsonschema_test.go [hunk 1]
```diff
@@ -254,6 +254,53 @@ func TestIsCredentialValidForJSONSchema_JsonSchemaCredential(t *testing.T) {
 		})
 	})
 
+	t.Run("2.2 The value of the credentialSchema property MUST always be set to [known json schema]", func(t *testing.T) {
```
Comment (andresuribe87) [inline credential/schema/vcjsonschema_test.go:257]: Can you clarify that the 2.2 refers to a specific test from teh spec test suite?

File: credential/schema/vcjsonschema_test.go [hunk 2]
```diff
@@ -254,6 +254,53 @@ func TestIsCredentialValidForJSONSchema_JsonSchemaCredential(t *testing.T) {
...
+				"id":        JSONSchemaCredentialSchemaID,
+				"type":      "NotJsonSchema",
+				"digestSRI": JSONSchemaCredentialDigestSRI,
+			}
+			err := IsCredentialValidForJSONSchema(cred, schema, JSONSchemaCredentialType)
+			assert.Error(t, err)
+			assert.ErrorContains(t, err, "credential schema's credential schema type<NotJsonSchema> does not match known type<JsonSchema>")
+		})
+
+		t.Run("credentialSchema missing digestSRI", func(t *testing.T) {
```
Comment (andresuribe87) [inline credential/schema/vcjsonschema_test.go:291]: Can you add a test where the `digestSRI`, when computed from the source, doesn't match the one that was given in the VC?

Comment (decentralgabe) [reply credential/schema/vcjsonschema_test.go:291]: yes

File: credential/schema/vcjsonschema_test.go (hunk 1 above)
Comment (decentralgabe) [reply credential/schema/vcjsonschema_test.go:257]: yes

**Review by andresuribe87 — COMMENTED** [review]
We will also need to validate that a VC that's setting the `credentialSchema.digestSRI` property, has a value which matches against the locally calculated digestSRI. This typically involves dereferencing `credentialSchema.id`, and then following the spec of SubResource Integrity. 

I would prefer that we do that in this PR, so that the same mechanism is used for every VC (instead of special casing VCs who's `type` property is `JsonSchemaCredential`). That said, I'd understand leaving that for another PR. Up to you.

**Comment by decentralgabe** [comment]
I'd rather take that in a separate PR, but opened an issue https://github.com/TBD54566975/ssi-sdk/issues/456

**Review by andresuribe87 — APPROVED** [review]

**Pull request merged**